src/snowflake/snowpark/_internal/analyzer/snowflake_plan_node.py

```diff
@@ -23,6 +24,13 @@
 class LogicalPlan:
     def __init__(self) -> None:
         self.children = []
+        self._id = str(uuid.uuid4())
+
+    def __eq__(self, other: "LogicalPlan") -> bool:
+        return isinstance(other, LogicalPlan) and (self._id == other._id)
+
+    def __hash__(self) -> int:
+        return hash(self._id)
```

Comment:
My understanding is that we are relying on this during traversal to find duplicate nodes, is that correct? I have seen usage patters where users have dataframe constructor class to generate specific dataframes defined by configurations given to the constructor class. My concern is that this approach will not detect dataframes that are duplicates - as in they embody the exact same query - but they will not get detected as duplicates because they have different UUIDs.

Reply:
I think your point makes sense, and we can generate hash based on query text. But then I need to move `_id` to `SnowflakePlan` instead of `LogicalPlan` because query is generated upon a Snowflake Plan is created

Reply:
done